PEP 622: make the binding semantics explicit (#1490)
python/peps: pep-0622.rst

          ...
  print(x, y)  # This works

> During failed pattern matches, some subpatterns may succeed. For example,

Reviewer: ```suggestion
During failed pattern matches, some sub-patterns may succeed. For example,
```

while matching the value ``[0, 1, 2]`` with the pattern ``(0, x, 1)``, the

> subpattern `x` may succeed if the list elements are matched from left to right.

Reviewer: ```suggestion
sub-pattern `x` may succeed if the list elements are matched from left to right.
```

The implementation may choose to either make persistent bindings for those
partial matches or not. User code including a `match` statement should not rely
on the bindings being made for a failed match, but also shouldn't assume that

> variables being unchanged by a failed match. This part of the behavior is

Reviewer: ```suggestion
variables are unchanged by a failed match. This part of the behavior is
```

left intentionally unspecified so different implementations can add
optimizations, and to prevent introducing semantic restrictions that could

> limit the extendability of this feature.

Reviewer: ```suggestion
limit the extensibility of this feature.
```

> Note that some of the pattern kinds below define more specific rules about when

Reviewer: ```suggestion
Note that some pattern types below define more specific rules about when
```
